examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt

```diff
@@ -106,6 +119,23 @@ class ReplayHistoryActivity : AppCompatActivity() {
         }
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        HistoryFilesActivity.selectedHistory?.let {
+            val replayHistoryMapper = ReplayHistoryMapper(Gson(), ReplayCustomEventMapper(), MapboxLogger)
+            val replayEvents = replayHistoryMapper.mapToReplayEvents(it)
+            navigationContext?.mapboxReplayer?.stop()
+            navigationContext?.mapboxReplayer?.clearEvents()
+            navigationContext?.mapboxReplayer?.pushEvents(replayEvents)
+            navigationContext?.mapboxReplayer?.seekTo(0.0)
+
+            // We have a new set of data, so reset the navigator
+            MapboxNativeNavigatorImpl.create(HandheldProfile(), MapboxLogger)
```

Comment:
There is offline discussion on how we should handle this. The only way we know to reset navigation after selecting history files, is to use an internal method. This call can also be triggered with a public function using MapboxNavigation.onDestroy